test/performance/benchmarks/reconciliation-delay/main.go

```diff
@@ -327,5 +328,12 @@ func checkSLA(results *vegeta.Metrics, expectedReadyServices float64) error {
 		return fmt.Errorf("SLA 2 failed. P95 latency is not in 0-25s time range: %s", results.Latencies.P95)
 	}
 
+	// SLA 3: making sure the defined vegeta rates/total requests are met
+	if results.Requests == uint64(expectedReadyServices) {
```

Comment:
It seems to me a bit counter-intuitive. Reason this should work is because each time we check ready==true we do metricResults.Add(&result) which implicitly increases requests.